pkg/common-controller/snapshot_controller.go

```diff
@@ -1591,3 +1602,31 @@ func (ctrl *csiSnapshotCommonController) checkAndSetInvalidSnapshotLabel(snapsho
 
 	return updatedSnapshot, nil
 }
+
+func (ctrl *csiSnapshotCommonController) getManagedByNode(snapshot *crdv1.VolumeSnapshot) (string, error) {
+	pv, err := ctrl.getVolumeFromVolumeSnapshot(snapshot)
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve PV from snapshot[%s]: %q", snapshot.Name, err)
+	}
+	if pv.Spec.NodeAffinity == nil {
+		klog.V(5).Infof("NodeAffinity not set for pv %s", pv.Name)
+		return "", nil
+	}
+	nodeSelectorTerms := pv.Spec.NodeAffinity.Required
+
+	nodes, err := ctrl.client.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("failed to get the list of nodes: %q", err)
+		return "", nil
```

Comment:
this should return an error rather? otherwise a wrong content will be created

Reply:
If we want to return an error here, we'll have to do that under the condition if distributed snapshotting is enabled. We'll probably have to pass a flag to this controller to identify that.

Right now, the snapshot_controller code path is same for both distributed and non distributed snapshotting and there's no distinction between these two scenarios. I thought we should keep it that way as we'll have only 1 centralized instance of this controller running even in case of distributed snapshotting. So this controller doesn't need to know if the snapshotter is being deployed on a per node basis or not. WDYT?

Reply:
Returning an error here now.